src/main/java/com/ldtteam/structures/blueprints/v1/BlueprintUtil.java

```diff
@@ -184,49 +169,49 @@ public static NBTTagCompound writeBlueprintToNBT(Blueprint schem)
         tag.setTag("palette", paletteTag);
 
         // Adding blocks
-        int[] blockInt = convertBlocksToSaveData(schem.getStructure(), schem.getSizeX(), schem.getSizeY(),
+        final int[] blockInt = convertBlocksToSaveData(schem.getStructure(), schem.getSizeX(), schem.getSizeY(),
           schem.getSizeZ());
         tag.setIntArray("blocks", blockInt);
 
         // Adding Tile Entities
-        NBTTagList finishedTes = new NBTTagList();
-        NBTTagCompound[] tes = schem.getTileEntities();
-        for (int i = 0; i < tes.length; i++)
+        final NBTTagList finishedTes = new NBTTagList();
+        final NBTTagCompound[] tes = Arrays.stream(schem.getTileEntities()).flatMap(Arrays::stream).flatMap(Arrays::stream).toArray(NBTTagCompound[]::new);
```

Comment:
Why double map and why stream from array to array?
And Stream.of(array)

Reply:
It's a dim array which we transform into a 1 dim list, so 1 flatMap per dim